src/gui/dialogs/unit_recruit.cpp

```diff
@@ -55,13 +57,74 @@ static std::string can_afford_unit(const std::string& text, const bool can_affor
 	return can_afford ? text : "<span color='#ff0000'>" + text + "</span>";
 }
 
+namespace
+{
+
+bool ci_search(const std::string& a, const std::string& b)
+{
+	return std::search(a.begin(), a.end(),
+	                   b.begin(), b.end(),
+	                   chars_equal_insensitive) != a.end();
+}
+
+} // end unnamed namespace
+
+// Compare unit_create::filter_text_change
+void unit_recruit::filter_text_changed(text_box_base* textbox, const std::string& text)
+{
+	window& window = *textbox->get_window();
+
+	listbox& list = find_widget<listbox>(&window, "recruit_list", false);
+
+	const std::vector<std::string> words = utils::split(text, ' ');
+
+	if(words == last_words_)
+		return;
+	last_words_ = words;
+
+	boost::dynamic_bitset<> show_items;
+	show_items.resize(list.get_item_count(), true);
+
+	if(!text.empty()) {
+		for(unsigned int i = 0; i < list.get_item_count(); i++) {
+			assert(i < recruit_list_.size());
+			const unit_type* type = recruit_list_[i];
+			if(!type) continue;
+
+			bool found = false;
+			for(const auto & word : words)
+			{
+				// Search for the name in the local language.
+				// In debug mode, also search for the English name.
+				found =
+				        (game_config::debug && ci_search(type->id(), word)) ||
```

Comment:
Wouldnt it search by type id rather than English name then?

Reply:
Well, yes. In debug mode it will search match against both the type id and the localized name. Shall I rephrase the comment?

Reply:
Yes.

Reply:
Done.